### ISSUE TITLE
Kernel 4.9

### ISSUE THREAD

**Issue by mterron** [KL-Yang/mbl-linux-4.0#7]
Any chance of supporting Kernel 4.9?

**Comment by KL-Yang** [comment]
Ewald in the WD forum has posted a 4.9 kernel (I had a little trouble to run it on the single bay MBL).
I have create a patch-4.9 branch, mostly a rebase with some minor fix of compiling errors.
I am currently running this kernel on my MBL to see how it goes.
If you are interested, I can provide a compiled package for testing.

**Comment by mterron** [comment]
Sure, a .deb would be great.

Thanks

**Comment by KL-Yang** [comment]
The 4.9.39 kernel tar-gz package can be found here.
https://drive.google.com/open?id=0B-PZDFHXqH6pQTdJZUg0WVQ5Qzg
The kernel is built on MBL itself running 4.9 kernel, but not sure how stable it is.

**Comment by mterron** [comment]
Any update on this? A fresh image based on Stretch with current kernels would be awesome to keep our little MBLs alive.

**Comment by goldstar611** [comment]
Unfortunately Debian Stretch has removed support for PPC architecture (leaving only ppc64el) so it's going to be increasingly harder to come by full images that are up to date. I too would be interested to know if anyone is tackling building stretch from source (either native or cross compile)

**Comment by mterron** [comment]
True, even a 'current' Jessie image would be useful. Anyway the MBL does a wonderful job as a pihole.

**Comment by goldstar611** [comment]
The thing is about Jessie, I was never able to get the WD-specific stuff working (I think due to /bin/dash being default shell and Apache complaining about conf files). If there's any documentation that you know of getting those working I wouldn't mind getting a fresh MBL Jessie image together.

**Comment by antrew** [comment]
If anybody is interested, I have written some scripts to cross-build the kernel (4.9) and Jessie in a Vagrantbox from scratch. Have a look at my fork: https://github.com/antrew/mbl-linux-4.0
It contains a README.md with instructions. Fell free to open an issue if something doesn't work.
